extensions/amp-story/1.0/test/test-amp-story.js

```diff
@@ -1567,5 +1568,86 @@ describes.realWin(
         expect(enableSpy).to.be.calledWith('story-load-first-page-only');
       });
     });
+
+    describe('lazy load non-critical extensions after first page is loaded', () => {
+      it('should install auto-ads if a config is provided', async () => {
+        const pages = await createStoryWithPages(2, ['cover', 'page-1']);
+        const extensionsFor = Services.extensionsFor(win);
+        element.appendChild(<amp-story-auto-ads></amp-story-auto-ads>);
+
+        const signals = new Signals();
+        pages[0].signals = () => signals;
+        story.buildCallback();
+        await story.layoutCallback();
+
+        const installSpy = env.sandbox.spy(
+          extensionsFor,
+          'installExtensionForDoc'
+        );
+
+        // Signal that the first page finished loading.
+        signals.signal(CommonSignals_Enum.LOAD_END);
```

Comment:
I don't think you need to call this manually, I see other tests doing things like

```
        await story.layoutCallback();
        await story.activePage_.element
          .signals()
          .whenSignal(CommonSignals_Enum.LOAD_END);
```

So maybe you could

```
        createStoryWithPages();
        append(amp-story-auto-ads);
        stub(installExtensionForDoc);

        await story.layoutCallback();
        await story.activePage_.element
          .signals()
          .whenSignal(CommonSignals_Enum.LOAD_END);

        expect(ampstoryautoads to be loaded);
```